src/variable_rate.jl

```diff
@@ -36,19 +36,19 @@ mutable struct VRDirectCBEventCache{T, RNG <: AbstractRNG}
     prev_time::T
     prev_threshold::T
     current_time::T
-    current_threshold::T  
-    total_rate_cache::T
+    current_threshold::T
+    total_rate::T
     rng::RNG
     variable_jumps::Tuple{Vararg{VariableRateJump}}
-    rate_funcs::Vector{Function}
-    affect_funcs::Vector{Function}
+    rate_funcs::Vector{FunctionWrappers.FunctionWrapper{T, Tuple{T, T, Any, T}}}
```

Comment:
Don’t use function wrappers. That is introducing an unneeded performance hit for the common case of only having a few jumps. Please see my comments I made when I gave you a review a while ago and use tuples and recursion like we do for the Direct aggregator.